### PR TITLE
Bug fixes in financial statements

### PR DESCRIPTION
**Pull request by the-praxs**
**Changelog** -

* Modified financial statement standard model fields to type `StrictFloat` and default period to `quarter`
* Add `ttm` to the `period` param in Polygon
* Fixed Polygon `period` param not working when value is `quarter`
* Modified FMP Stock Insider to use `limit` instead of `page` to query data
* Modified Intrinio balance sheet model to use `tag` instead of `name` from returned data for field name
* Fixed Intrinio Income Statement model fetching `balance_sheet_statement` endpoint instead of `income_statement`
* Add additional calculated data in Intrinio financials 
* Fixed Intrinio financial statement models to return the same data as other providers

### CODE REVIEW

File: openbb_platform/providers/polygon/openbb_polygon/models/balance_sheet.py [hunk 1]
```diff
@@ -20,8 +21,13 @@ class PolygonBalanceSheetQueryParams(BalanceSheetQueryParams):
     Source: https://polygon.io/docs/stocks#!/get_vx_reference_financials
     """
 
-    __alias_dict__ = {"symbol": "ticker", "period": "timeframe"}
+    __alias_dict__ = {"symbol": "ticker"}
 
+    period: Optional[Literal["annual", "quarter", "ttm"]] = Field(
```
Comment (deeleeramone) [inline openbb_platform/providers/polygon/openbb_polygon/models/balance_sheet.py:26]: This doesn't appear to work, all three periods return exactly the same thing:

Screenshot 2023-10-19 at 11 47 57 AM

Screenshot 2023-10-19 at 11 54 16 AM

Screenshot 2023-10-19 at 11 55 52 AM

**Review by deeleeramone — CHANGES_REQUESTED** [review]

File: openbb_platform/providers/intrinio/openbb_intrinio/models/balance_sheet.py [hunk 2]
```diff
@@ -85,9 +85,14 @@ def extract_data(
...
-            # TODO: Fix quarter range after Intrinio's response
-            for quarter in range(1, 4):
-                url = f"{base_url}/fundamentals/{url_params}-Q{quarter}/{statement_param}?api_key={api_key}"
+            quarter_range = get_quarter_range(query.year)
+
+            for quarter in quarter_range:
+                # TODO: Check back in sometime when Intrinio fixes their API/provides better documentation
+                if quarter == 4 and query.type == "reported":
+                    url = f"{base_url}/fundamentals/{url_params}-FY/{statement_param}?api_key={api_key}"
+                else:
```
Comment (deeleeramone) [inline openbb_platform/providers/intrinio/openbb_intrinio/models/balance_sheet.py:94]: This does not appear to work. All periods return: 

```
OpenBBError: Intrinio endpoint doesn't exist
```

File: openbb_platform/providers/polygon/openbb_polygon/models/cash_flow.py [hunk 3]
```diff
@@ -113,6 +119,7 @@ def extract_data(
         api_key = credentials.get("polygon_api_key") if credentials else ""
 
         base_url = "https://api.polygon.io/vX/reference/financials"
+        query.period = "quarterly" if query.period == "quarter" else query.period
```
Comment (deeleeramone) [inline openbb_platform/providers/polygon/openbb_polygon/models/cash_flow.py:122]: Key Error on cashflow statement.

```
In [11]: obb.stocks.fa.cash("AAPL", provider="polygon", period= "ttm").to_df().transpose()
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File ~/GitHub/OpenBBTerminal/openbb_platform/platform/provider/openbb_provider/query_executor.py:87, in QueryExecutor.execute(self, provider_name, model_name, params, credentials, **kwargs)
     86 try:
---> 87     return fetcher.fetch_data(params, filtered_credentials, **kwargs)
     88 except Exception as e:

File ~/GitHub/OpenBBTerminal/openbb_platform/platform/provider/openbb_provider/abstract/fetcher.py:55, in Fetcher.fetch_data(cls, params, credentials, **kwargs)
     54 data = cls.extract_data(query=query, credentials=credentials, **kwargs)
---> 55 return cls.transform_data(query=query, data=data, **kwargs)

File ~/GitHub/OpenBBTerminal/openbb_platform/providers/polygon/openbb_polygon/models/cash_flow.py:143, in PolygonCashFlowStatementFetcher.transform_data(query, data, **kwargs)
    140 for item in data:
    141     sub_data = {
    142         key: value["value"]
--> 143         for key, value in item["financials"]["cash_flow_statement"].items()
    144     }
    145     sub_data["date"] = item["start_date"]

KeyError: 'cash_flow_statement'
```

**Review by deeleeramone — CHANGES_REQUESTED** [review]

File: openbb_platform/providers/intrinio/openbb_intrinio/models/cash_flow.py [hunk 4]
```diff
@@ -93,16 +99,21 @@ def extract_data(
...
 
         elif query.period == "quarter":
-            # TODO: Fix quarter range after Intrinio's response
-            for quarter in range(1, 4):
-                url = f"{base_url}/fundamentals/{url_params}-Q{quarter}/{statement_param}?api_key={api_key}"
+            quarter_range = get_quarter_range(query.year)
+
+            for quarter in quarter_range:
+                # TODO: Check back in sometime when Intrinio fixes their API/provides better documentation
+                if quarter == 4 and query.type == "reported":
```
Comment (deeleeramone) [inline openbb_platform/providers/intrinio/openbb_intrinio/models/cash_flow.py:113]: This does not appear to work:

Screenshot 2023-10-19 at 12 24 00 PM


Changing the symbol to "JPM" creates an error:

```
OpenBBError: Intrinio endpoint doesn't exist
```

Tickers that do return, return all NaNs for Q2 and Q3

Screenshot 2023-10-19 at 12 28 59 PM

**Review by deeleeramone — CHANGES_REQUESTED** [review]

File: openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def extract_data(
         api_key = credentials.get("intrinio_api_key") if credentials else ""
 
         base_url = "https://api-v2.intrinio.com"
-        url_params = f"{query.symbol}-balance_sheet_statement-{query.year}"
+        url_params = f"{query.symbol}-income_statement-{query.year}"
```
Comment (deeleeramone) [inline openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py:74]: symbol="JPM" returns:

```
OpenBBError: Intrinio endpoint doesn't exist
```

It does exist though, so the URLs are not generating correctly:

```
https://api-v2.intrinio.com/fundamentals/JPM-income_statement-2023-Q2/standardized_financials?
```

File: openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py [hunk 6]
```diff
@@ -95,30 +100,28 @@ def transform_data(
...
+            for sub_item in item[data_key]:
+                field_name = alias_generators.to_snake(sub_item[tag_key]["name"])
+                sub_dict[field_name] = float(sub_item["value"])
 
             sub_dict["date"] = item["fundamental"]["end_date"]
             sub_dict["period"] = item["fundamental"]["fiscal_period"]
             sub_dict["cik"] = item["fundamental"]["company"]["cik"]
             sub_dict["symbol"] = item["fundamental"]["company"]["ticker"]
 
+            # Intrinio does not return Q4 data in reported_financials endpoint
```
Comment (deeleeramone) [inline openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py:118]: This is not true these both return data:

```
https://api-v2.intrinio.com/fundamentals/JPM-income_statement-2022-Q4/standardized_financials?
https://api-v2.intrinio.com/fundamentals/AAPL-income_statement-2022-Q4/standardized_financials?
```

**Review by deeleeramone — CHANGES_REQUESTED** [review]

File: openbb_platform/providers/polygon/openbb_polygon/models/cash_flow.py (hunk 3 above)
Comment (the-praxs) [reply openbb_platform/providers/polygon/openbb_polygon/models/cash_flow.py:122]: `ttm` is only for the balance sheet. Removing these from Polygon for cash flow and income statement.

image

File: openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py (hunk 6 above)
Comment (the-praxs) [reply openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py:118]: It says `reported_financials` also in the comment.

Comment (deeleeramone) [reply openbb_platform/providers/intrinio/openbb_intrinio/models/income_statement.py:118]: They are all accepted through the same endpoint.

Screenshot 2023-10-19 at 6 52 17 PM

**Review by IgorWounds — APPROVED** [review]

**Pull request merged**